system_intelligence/ram_info.py

```diff
@@ -1,190 +1,229 @@
-"""Functions to query system's RAM."""
-
 import logging
 import os
 import subprocess
 import typing as t
 from xml.etree import ElementTree as ET
-
 import psutil
-from rich.console import Console
 from rich import print
+from sys import platform
 
+from .base_info import BaseInfo
 from .util.process_util import is_process_accessible
-from .util.rich_util import create_styled_table
 from .util.unit_conversion_util import bytes_to_hreadable_string, hz_to_hreadable_string
 
 _LOG = logging.getLogger(__name__)
 
 try:
     import pyudev
-
     pyudev.Context()
 except ImportError:
     pyudev = None
-    print('[bold yellow]Unable to import package pyudev. HDD information may be limited.')
-
-RAM_TOTAL = psutil is not None
-
-
-def query_ram_total() -> t.Optional[int]:
-    """Get information about total available RAM."""
-    if not RAM_TOTAL:
-        return None
-
-    return psutil.virtual_memory().total
-
-
-def query_ram(sudo: bool = False, **kwargs) -> t.Mapping[str, t.Any]:
-    """Get all available information about RAM."""
-    total_ram = query_ram_total()
-    ram_banks, ram_cache = query_ram_banks_cache(sudo=sudo, **kwargs)
-    ram: t.Dict[str, t.Any] = {'total': total_ram, 'banks': {}, 'cache': {}}
-    if ram_banks:
-        ram['banks'] = ram_banks
-    if ram_cache:
-        ram['cache'] = ram_cache
-
-    return ram
-
-
-def query_ram_banks_cache(sudo: bool = False, **_) \
+    print('[bold yellow]Unable to import package pyudev. RAM information may be limited.')
+
+
+class RamInfo(BaseInfo):
+    """
+    Bla
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.RAM_TOTAL = psutil is not None
+        self.OS = platform
```

Comment:
Thinking of making `OS` a variable of `BaseInfo`. Since this information is not only needed here, but elsewhere (for example CPU)